modules/sfp_opencorporates.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_opencorporates
+# Purpose:      SpiderFoot plug-in for retrieving company location and previous
+#               company names from OpenCorporates.
+#
+# Author:      Brendan Coles <bcoles@gmail.com>
+#
+# Created:     2018-10-21
+# Copyright:   (c) Brendan Coles 2018
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import re
+import urllib
+from sflib import SpiderFoot, SpiderFootPlugin, SpiderFootEvent
+
+class sfp_opencorporates(SpiderFootPlugin):
+    """OpenCorporates:Passive,Footprint,Investigate:Search Engines::Look up company information from OpenCorporates."""
+
+    # Default options
+    opts = {
+        'confidence': 100
+    }
+
+    # Option descriptions
+    optdescs = {
+        'confidence': "Confidence that the search result objects are correct (numeric value between 0 and 100)."
+    }
+
+    results = dict()
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.__dataSource__ = "OpenCorporates"
+        self.results = dict()
+
+        for opt in userOpts.keys():
+            self.opts[opt] = userOpts[opt]
+
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return ["COMPANY_NAME"]
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["AFFILIATE_COMPANY_NAME", "GEOINFO"]
```

Comment:
Can we get a bit more, like the human names of company directors?

Reply:
This can be retrieved, at the cost of an extra HTTP request. Only some companies have listed officers.

The problem is, even when `confidence` is set to `100`, it's still possible that two companies in two different countries will have the exact same name.

The module fires on `COMPANY_NAME` event, but without any associated location data. This could result in reporting a bunch of `HUMAN_NAME` objects for a completely different company.

Reply:
Updated to retrieve officer names.